ApiBundle/Transformer/NodeTransformer.php

```diff
@@ -152,6 +150,37 @@ public function transform($node)
     }
 
     /**
+     * Get a preview link
+     * 
+     * @param string             $scheme
+     * @param SiteAliasInterface $alias
+     * @param string             $encryptedId
+     * 
+     * @return FacadeInterface|LinkFacade
+     */
+    protected function getPreviewLink($scheme, $alias, $encryptedId)
+    {
+        $previewLink = array(
+            'name' => $alias->getDomain(),
+            'link' => ''
+        );
+
+        if (is_null($scheme) || SchemeableInterface::SCHEME_DEFAULT == $scheme) {
+            $scheme = $alias->getScheme();
+        }
+        $domain = $scheme . '://' . $alias->getDomain();
+
+        if ($alias->getPrefix() != "") {
+            $previewLink['name'] .= '/' . $alias->getPrefix();
+            $previewLink['link'] = $domain . $this->generateRoute('node_preview_with_prefix', array('languagePrefix' => $alias->getPrefix(), 'token' => $encryptedId), UrlGeneratorInterface::ABSOLUTE_PATH);
+        } else {
```

Comment:
You could maybe suppress else by defining two default values for generateRoute out of "if", override in "if" and initialize link out of if.